### PR TITLE
Updated Release instructions

### PR DESCRIPTION
**Pull request by devjgm**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/1964)

### CODE REVIEW

File: doc/cutting-a-release.md [hunk 1]
```diff
@@ -81,41 +89,54 @@ git commit -m"Create ${RELEASE}.x release branch" .
 git push --set-upstream origin ${RELEASE}.x
 ```
 
+NOTE: No code review or Pull Request is needed as part of this step.
```
Comment (coryan) [inline doc/cutting-a-release.md]: Do you want that to be `**NOTE:**` so it is in **BOLD** ?

File: doc/cutting-a-release.md [hunk 2]
```diff
@@ -81,41 +89,54 @@ git commit -m"Create ${RELEASE}.x release branch" .
...
+send the PR for review against `master`. For an example, look at
+[#1962](https://github.com/googleapis/google-cloud-cpp/pull/1962)
 
 ## Create a pre-release tag
 
 Create a pre-release using
 [GitHub](https://github.com/googleapis/google-cloud-cpp/releases/new).
-Make sure your reference the `v0.N.x` branch, and you check the `pre-release`
-checkbox.
+Make sure you reference the `v0.N.x` branch, set a tag name like `v0.N.0-pre1`,
```
Comment (coryan) [inline doc/cutting-a-release.md]: Yikes, thanks for the s/your/you/ change.

File: doc/cutting-a-release.md [hunk 3]
```diff
@@ -81,41 +89,54 @@ git commit -m"Create ${RELEASE}.x release branch" .
...
+and you check the `pre-release` checkbox.
```
Comment (coryan) [inline doc/cutting-a-release.md]: should that be "and check the `pre-release` checkbox."?

File: doc/cutting-a-release.md (hunk 1 above)
Comment (devjgm) [reply doc/cutting-a-release.md]: yep. I like that. Done.

File: doc/cutting-a-release.md (hunk 3 above)
Comment (devjgm) [reply doc/cutting-a-release.md]: Yep.

**Pull request merged**